test/EncumberableTokenFactory.t.sol

```diff
@@ -6,13 +6,21 @@ import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import { IERC20Metadata } from "@openzeppelin/contracts/interfaces/IERC20Metadata.sol";
 
 contract EncumberableTokenFactoryTest is Test {
+    event WrapperDeployed(address indexed underlyingToken, address indexed wrapperToken);
+
     EncumberableTokenFactory public wrapperFactory;
     ERC20 public erc20;
     address public wrappedToken;
 
     function setUp() public {
         wrapperFactory = new EncumberableTokenFactory();
         erc20 = new ERC20("TEST TOKEN", "TTKN");
+
+        vm.expectEmit(true, true, true, true);
+        emit WrapperDeployed(
+            address(erc20),
+            wrapperFactory.getDeploymentAddress(address(erc20))
+        );
```

Comment:
Think it's better to test this within its own test function.

Reply:
moved to a separate test function